cmd/althea/cmd/root.go

```diff
@@ -109,34 +196,166 @@ func Execute(rootCmd *cobra.Command) error {
+		// ValidateChainID will make sure the configured chain id adheres to strings like althea_1234-1
+		ValidateChainID(InitCmd(althea.ModuleBasics, althea.DefaultNodeHome)),
+		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, althea.DefaultNodeHome),
 		althea.MigrateGenesisCmd(),
-		GenTxCmd(althea.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, althea.DefaultNodeHome),
+		genutilcli.GenTxCmd(althea.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, althea.DefaultNodeHome),
 		genutilcli.ValidateGenesisCmd(althea.ModuleBasics),
 		AddGenesisAccountCmd(althea.DefaultNodeHome),
 		tmcli.NewCompletionCmd(rootCmd, true),
 		testnetCmd(althea.ModuleBasics, banktypes.GenesisBalancesIterator{}),
-		debug.Cmd(),
+		debug.Cmd(),  // Output useful info about keys without writing Go
+		config.Cmd(), // Set config options one by one
 	)
 
-	server.AddCommands(rootCmd, althea.DefaultNodeHome, newApp, createSimappAndExport, addModuleInitFlags)
+	// The ethermint server commands perform a lot of modifications on top of the base ones, notably setting up the
+	// EVM JSONRPC server, tx indexer, and some various improvements like closing the DB automatically
+	ethermintserver.AddCommands(rootCmd, althea.DefaultNodeHome, newApp, createSimappAndExport, addModuleInitFlags)
+
+	rootCmd.AddCommand(ethermintserver.NewIndexTxCmd())
 
 	// add keybase, auxiliary RPC, query, and tx child commands
 	rootCmd.AddCommand(
 		rpc.StatusCommand(),
+		// queryCommand registers the query subcommands by looking at ModuleBasics
 		queryCommand(),
+		// txCommand registers the tx subcommands by looking at ModuleBasics
 		txCommand(),
-		keys.Commands(althea.DefaultNodeHome),
+		// Adds the same commands as sdk keys.Commands(), but enables dry run and sets the default keytype to eth_secp256k1
+		// keys.Commands(althea.DefaultNodeHome),
+		ethermintclient.KeyCommands(althea.DefaultNodeHome),
 		Commands(althea.DefaultNodeHome),
 	)
 }
+
+// InitCmd returns a command that initializes all files needed for Tendermint
+// and the respective application.
+// Note that this is mostly a copy of the default InitCmd found in genutil, however we need to overwrite the default
+// chain id to one which will not cause a panic in ethermint/x/evm/genesis.go
+func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "init [moniker]",
+		Short: "Initialize private validator, p2p, genesis, and application configuration files",
+		Long:  `Initialize validators's and node's configuration files.`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			cdc := clientCtx.Codec
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+			config.SetRoot(clientCtx.HomeDir)
+
+			chainID, _ := cmd.Flags().GetString(flags.FlagChainID)
```

Comment:
We are unable to import the Canto/Ethermint InitCmd implementations primarily because of the chain id defaults they overwrite with, which are always [canto/evmos]_####-#